thunder/executors/nvfuserex_impl.py

```diff
@@ -1214,9 +1214,8 @@ def _reshape_check(a: TensorProxy, shape: list[int]) -> bool:
 
 def reshape(a: TensorProxy, shape: list[int], *, fd: FusionDefinition, lc_to_nv_map: dict) -> Any:
     nv_a = getnv(a, fd, lc_to_nv_map)
-    nv_shape = getnv(shape, fd, lc_to_nv_map)
```

Comment:
I see what you are trying to do here. I think we need to check that `shape` does not contain `NumberProxy` and only do this simplification then.

CI failures looks strange, seems like an infra issue?

Reply:
I'm probably missing something. Isn't shape of type list[int] as defined in the signature?

Reply:
yeah, that annotation needs to be updated. 
But if we are to support dynamic shape, we'll update it as `shape: list[int, NumberProxy]`.

Reply:
to be specific, I'm saying that we should do this instead:
```
    if any(map(lambda x: isinstance(x, Proxy), shape)):
        nv_shape = getnv(shape, fd, lc_to_nv_map)
    else: 
        nv_shape = shape 
```

extra credit if we also fix the annotation like the actual prim: https://github.com/Lightning-AI/lightning-thunder/blob/f70b0fe22d27a2e2a0a6ac31cf2c52c9d4e4e35d/thunder/core/prims.py#L3158

Reply:
could also use a comment as, why we would want to use `shape` directly in the code.

Reply:
Do we need to make a change now for dynamic shapes? `broadcast_in_dim` should have identical `shape` usage.

Reply:
The only reason shape was not originally used, directly, was because nvFuser previously required using `define_scalar()` to add a scalar constant.  That behavior was changed a long time ago and did not get propagated to `reshape`, for whatever reason, but did make it to the `broadcast_in_dim` definition.

Reply:
As long as you are passing CIs, you don't have to. I don't mind clean up afterwards.

But I'm a bit suspicious that you are not going to pass CI, because we happen to have a reshape test case with symbolic values: https://github.com/Lightning-AI/lightning-thunder/blob/f70b0fe22d27a2e2a0a6ac31cf2c52c9d4e4e35d/thunder/tests/test_jit_general.py#L1124-L1125

Reply:
actually, that one just checks for cache hit... so even though nvfuser recompiles I think you are still good.